cmd/kube-proxy/app/server_linux.go

```diff
@@ -109,10 +110,12 @@ func isIPTablesBased(mode proxyconfigapi.ProxyMode) bool {
 func (s *ProxyServer) platformCheckSupported(ctx context.Context) (ipv4Supported, ipv6Supported, dualStackSupported bool, err error) {
 	logger := klog.FromContext(ctx)
 
+	_, errIPv6 := os.Stat("/proc/net/if_inet6")
```

Comment:
another option is to try to enable a socket in ipv6 only, that will guarantee that the applications can actually work in ipv6 

```
func isIPv6Capable() bool {
	udp6Conn, err := net.ListenUDP("udp6", &net.UDPAddr{IP: net.IPv6zero, Port: 0})
	if err != nil {
                 return false
	}
        udp6Conn.Close()
        return true
}
```

Reply:
> guarantee

Well, the problem is that there are lots of different knobs that disable different aspects of IPv6. I don't think it's really possible to _completely reliably_ decide "is the kernel going to accept all of our IPv6 rules" other than by actually trying to create all of our IPv6 rules. It's all just heuristic, so I think it's worth going with a simpler heuristic over a more complicated one.

(Well... actually, we _could_ do a knftables `Check()` to test an IPv6 ruleset, but that would require us to write up a special test transaction that has one of every kind of ipv6 rule we might create later on... I hope we don't actually need to do that, but it's always a possibility if this current PR proves to be insufficient.)

Reply:
I think opening the listener is fine as well, if you folks think it is a good way to check. Dan nailed the current problem I've faced locally, which is at some point some of the nftables rules were rejected by Kernel, but I could still create inet6 tables, etc.

So every time we expand nftables proxier for some new kind of rule/module (let's say, some ratelimit, some new NAT module, etc) we would need to test this as well which may be...hard